firedrake/function.py

```diff
@@ -413,6 +415,14 @@ def assign(self, expr, subset=None):
         """
         if expr == 0:
             self.dat.zero(subset=subset)
+        # Assign literal collections to vector and tensor valued functions in R.
+        elif (isinstance(expr, collections.abc.Container)
+              and self.ufl_element().family() == "Real"):
```

Comment:
```suggestion
        elif self.ufl_element().family() == "Real":
```

I think the extra `isinstance` check is overkill. Also won't `f.assign(non_vector_value)` fail with it when we would expect it to work?